client/src/modules/display_metadata/display_metadata.js

```diff
@@ -0,0 +1,208 @@
+  };
+
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  function toggleFilter() {
+    vm.filterEnabled = !vm.filterEnabled;
+    vm.gridOptions.enableFiltering = vm.filterEnabled;
+    vm.gridApi.core.notifyDataChange(uiGridConstants.dataChange.COLUMN);
+  }
+
+  // bind methods
+  vm.edit = edit;
+  vm.remove = remove;
+
+  function edit(data) {
+    if (!$state.params.patient) {
+      $state.go('display_metadata.edit', { id : data.data_collector_management_id, uuid : data.uuid });
+    } else if ($state.params.patient) {
+      $state.go('display_metadata.patientEdit', {
+        id : data.data_collector_management_id,
+        uuid : data.uuid,
+        patient : $state.params.patient,
+      });
+    }
+  }
+
+  function remove(uuid) {
+    ModalService.confirm('FORM.DIALOGS.CONFIRM_DELETE')
+      .then(() => {
+        DisplayMetadata.delete(uuid)
+          .then(() => {
+            Notify.success('FORM.INFO.DELETE_SUCCESS');
+            loadGrid();
+          })
+          .catch(Notify.handleError);
+      });
+  }
+
+  function handleError(error) {
+    vm.hasError = true;
+    Notify.handleError(error);
+  }
+
+  // load user grid
+  function loadGrid() {
+    toggleLoadingIndicator();
+    // Reuse cash unless patient data is consulted
+    if (cache.collector && !vm.collectorId && !$state.params.patient) {
+      vm.collectorId = cache.collector.id;
+    }
+    const changesLength = Object.keys(vm.changes).length;
+
+    if (cache.changes && !changesLength) {
+      vm.changes = cache.changes;
+    }
+
+    if (!$state.params.id && !vm.collectorId) {
+      search();
+    } else if (vm.collectorId && !$state.params.id) {
+      vm.hasError = false;
+      vm.loading = true;
+
+      SurveyForm.read(null, { data_collector_management_id : vm.collectorId })
+        .then((survey) => {
+          vm.filterElements = DisplayMetadata.displayFilters(survey, vm.changes);
+
+          vm.params = {
+            data_collector_management_id : vm.collectorId,
+            changes : vm.changes,
+          };
+
+          return DisplayMetadata.read(null, vm.params);
+        })
+        .then((data) => {
+          vm.gridOptions.columnDefs = data.columns;
+          vm.gridOptions.data = data.surveyData;
+
+          return DataCollectorManagement.read(vm.collectorId);
+        })
+        .then((collector) => {
+          cache.collector = collector;
+          cache.changes = vm.changes;
+          vm.collector = collector;
+        })
+        .catch(handleError)
+        .finally(toggleLoadingIndicator);
+
+    } else if ($state.params.id && $state.params.patient) {
+      vm.hasError = false;
+      vm.loading = true;
+      vm.collectorId = $state.params.id;
+      // View Patient Form Data
+      vm.params = {
+        data_collector_management_id : $state.params.id,
+        patient_uuid : $state.params.patient,
+        changes : vm.changes,
+      };
+
+      DisplayMetadata.read(null, vm.params)
```

Comment:
See if you can do a similar `Promise.all()` system here.

Reply:
![image](https://user-images.githubusercontent.com/8035588/64969611-26dc9980-d89c-11e9-9f10-a13b2b830506.png)
The displauing became very very too slow